demo/.strapi-app/scripts/seed.js

```diff
@@ -0,0 +1,306 @@
+      blockCopy.files = existingAndUploadedFiles;
+      // Push the updated block
+      updatedBlocks.push(blockCopy);
+    } else {
+      // Just push the block as is
+      updatedBlocks.push(block);
+    }
+  }
+
+  return updatedBlocks;
+}
+
+async function importArticles() {
+  for (const article of articles) {
+    const cover = await checkFileExistsBeforeUpload([`${article.slug}.jpg`]);
+    const updatedBlocks = await updateBlocks(article.blocks);
+
+    await createEntry({
+      model: 'article',
+      entry: {
+        ...article,
+        cover,
+        blocks: updatedBlocks,
+        // Make sure it's not a draft
+        publishedAt: Date.now(),
+      },
+    });
+  }
+}
+
+async function importGlobal() {
+  const favicon = await checkFileExistsBeforeUpload(['favicon.png']);
+  const shareImage = await checkFileExistsBeforeUpload(['default-image.png']);
+  return createEntry({
+    model: 'global',
+    entry: {
+      ...global,
+      favicon,
+      // Make sure it's not a draft
+      publishedAt: Date.now(),
+      defaultSeo: {
+        ...global.defaultSeo,
+        shareImage,
+      },
+    },
+  });
+}
+
+async function importAbout() {
+  const updatedBlocks = await updateBlocks(about.blocks);
+
+  await createEntry({
+    model: 'about',
+    entry: {
+      ...about,
+      blocks: updatedBlocks,
+      // Make sure it's not a draft
+      publishedAt: Date.now(),
+    },
+  });
+}
+
+async function importCategories() {
+  for (const category of categories) {
+    await createEntry({ model: 'category', entry: category });
+  }
+}
+
+async function importAuthors() {
+  for (const author of authors) {
+    const avatar = await checkFileExistsBeforeUpload([author.avatar]);
+
+    await createEntry({
+      model: 'author',
+      entry: {
+        ...author,
+        avatar,
+      },
+    });
+  }
+}
+
+async function importSeedData() {
+  // Allow read of application content types
+  await setPublicPermissions({
+    article: ['find', 'findOne'],
+    category: ['find', 'findOne'],
+    author: ['find', 'findOne'],
+    global: ['find', 'findOne'],
+    about: ['find', 'findOne'],
+  });
+
+  // Create all entries
+  await importCategories();
+  await importAuthors();
+  await importArticles();
+  await importGlobal();
+  await importAbout();
+}
+
+async function createApiTokens(app) {
+  const apiTokenService = app.service('admin::api-token');
+
+  const tokenTypes = [
+    { name: 'Full Access Token', type: 'full-access' },
+    { name: 'Read Only Token', type: 'read-only' },
+  ];
+
+  for (const tokenType of tokenTypes) {
+    const tokenDetails = {
+      name: tokenType.name,
+      description: `Token for ${tokenType.type} access`,
+      type: tokenType.type,
+      lifespan: null,
+    };
+
+    // Create the token
+    let token = await apiTokenService.create({
+      ...tokenDetails,
+    });
+
+    console.log(
+      `Created API Token ${token.name} (${token.type}) with access key: ${token.accessKey}`
```

Comment:
The tokens are logged there, but how do you envision the setup process to display again (reset) the tokens every time you want to use the demo app?

Reply:
Since you merged your PR already, I'll add something to echo them to .env automatically, and update your scripts to look there

That way any time the project is wiped/reseeded it will work

Reply:
that's what `pnpm demo:seed:clean` is intended to be

Reply:
Doesn't help I haven't noticed those changes, thanks :sweat_smile: